app/src/main/java/com/google/droidjump/achievements_data/AchievementsAdapter.java

```diff
@@ -0,0 +1,179 @@
+    private List<RecyclerViewItem> items;
+    private List<String> sectionNames;
+
+    public AchievementsAdapter(AchievementBuffer achievementBuffer, FragmentActivity activity) {
+        this.achievementBuffer = achievementBuffer;
+        this.activity = activity;
+        fillViewItems();
+    }
+
+    private void fillViewItems() {
+        items = new ArrayList<>();
+        sectionNames = new ArrayList<>();
+        for (int i = 0; i < achievementBuffer.getCount(); i++) {
+            if (i == 0 && achievementBuffer.get(i).getState() == Achievement.STATE_UNLOCKED) {
+                items.add(new RecyclerViewItem(ItemType.SECTION_NAME, sectionNames.size()));
+                sectionNames.add(String.valueOf(R.string.unlocked_section_name));
+            } else if (i == 0 || !(achievementBuffer.get(i).getState() == Achievement.STATE_UNLOCKED) && achievementBuffer.get(i - 1).getState() == Achievement.STATE_UNLOCKED) {
+                items.add(new RecyclerViewItem(ItemType.SECTION_NAME, sectionNames.size()));
+                sectionNames.add(String.valueOf(R.string.locked_section_name));
+            }
+            items.add(new RecyclerViewItem(ItemType.ACHIEVEMENT, i));
+        }
+    }
+
+    @NonNull
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        if (viewType == 0) {
+            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.section, parent, false);
+            return new SectionViewHolder(view);
+        }
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.achievement_item, parent, false);
+        return new AchievementViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder mholder, int position) {
+        if (items.get(position).getType() == ItemType.SECTION_NAME) {
+            SectionViewHolder holder = (SectionViewHolder) mholder;
+            holder.getSectionText().setText(sectionNames.get(items.get(position).getListPosition()));
+            return;
+        }
+        Achievement achievement = achievementBuffer.get(items.get(position).getListPosition());
+        AchievementViewHolder holder = (AchievementViewHolder) mholder;
+        holder.getDescription().setText(achievement.getDescription());
+        holder.getName().setText(achievement.getName());
+        ImageView icon = holder.getIcon();
+        ImageManager manager = ImageManager.create(activity);
+        switch (achievement.getState()) {
+            case Achievement.STATE_UNLOCKED:
+                manager.loadImage(icon, achievement.getUnlockedImageUri());
+                break;
+            case Achievement.STATE_REVEALED:
+                manager.loadImage(icon, achievement.getRevealedImageUri());
+                break;
+            case Achievement.STATE_HIDDEN:
+                break;
+        }
+    }
+
+    @Override
+    public int getItemViewType(int position) {
```

Comment:
Can this return an enum or a constant? Currently 0 and 1 does not mean anything.